UsersApp/AppDelegate/AppDelegate.swift

```diff
@@ -1,30 +1,69 @@
 import UIKit
-
+import GoogleSignIn
 
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
-
+    
     var window: UIWindow?
     
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         
         NetworkMonitor.shared.initiateConnectivityTracking()
         
+        GIDSignIn.sharedInstance.restorePreviousSignIn { user, error in
+            if let user = user {
+                print("Successfully restored previous sign-in for user: \(user.profile?.email ?? "")")
+                self.showSignedInState(for: user)
+            } else if let error = error {
+                print("Error restoring previous sign-in: \(error.localizedDescription)")
+                self.showSignedOutState()
+            } else {
+                print("No previous user found")
+                self.showSignedOutState()
+            }
+        }
```

Comment:
Create a Login service class to manage the Auth part of the app, call function from service here.
This is to separate logic and functionality. App delegate should not have a lot of code for different tasks